### PR TITLE
Add implicit type conversion to Maybe

### PR DESCRIPTION
**Pull request by erickotsuka**
Addressing issue #37.

Some suggestions on what to do about the extension `public static Maybe<T> ToMaybe<T>(this T value)` would be nice.

### CODE REVIEW

**Comment by alercunha** [comment]
We should also update the README file to incorporate an example of implicit conversion. My suggestion is that we do that using a method that is the more likely scenario. Something like:

```
public Maybe<int> GetValue()
{
   // calculate
   return 1;
}
```

**Review by alercunha — CHANGES_REQUESTED** [review]
We should also update the README file to incorporate an example of implicit conversion. My suggestion is that we do that using a method that is the more likely scenario. Something like:

```
public Maybe<int> GetValue()
{
   // calculate
   return 1;
}
```

File: Maybe.Test/MaybeTests.cs [hunk 1]
```diff
@@ -328,5 +329,45 @@ public void OrMaybe_Func_MaybeNullArgument_ShouldThrow()
...
+
+            subject.Value.Should().Be(value);
+        }
+
+        public static TheoryData<object> NonNullData()
+        {
+            return new TheoryData<object>()
+            {
+                '1',
+                1,
```
Comment (spliner) [inline Maybe.Test/MaybeTests.cs:354]: Let's add a test with a `0` so we can ensure that `if (value == null)` line doesn't get changed to a `if (value == default)`.

**Comment by alercunha** [comment]
Resolves #37

**Review by alercunha — APPROVED** [review]

File: Maybe/Maybe.csproj [hunk 2]
```diff
@@ -11,7 +11,7 @@
     <RepositoryUrl>https://github.com/zbra-solutions/maybe</RepositoryUrl>
     <RepositoryType>git</RepositoryType>
     <PackageLicenseExpression>MIT</PackageLicenseExpression>
-    <Version>0.2.0</Version>
+    <Version>0.3.0</Version>
```
Comment (spliner) [inline Maybe/Maybe.csproj:14]: Let's update to 1.0.0.

**Review by spliner — CHANGES_REQUESTED** [review]

File: Maybe/Maybe.csproj (hunk 2 above)
Comment (alercunha) [reply Maybe/Maybe.csproj:14]: Ship it!

File: Maybe.Test/MaybeTests.cs [hunk 3]
```diff
@@ -328,5 +329,46 @@ public void OrMaybe_Func_MaybeNullArgument_ShouldThrow()
...
+                0,
+                1.0,
+                DateTime.Now,
+                new List<string>(),
+                1.ToMaybe(),
```
Comment (dgmagno) [inline Maybe.Test/MaybeTests.cs:359]: We also need to test nullable structs

File: Maybe/Maybe.csproj (hunk 2 above)
Comment (d-nery) [reply Maybe/Maybe.csproj:14]: I believe this should be 0.3.0, we should move to 1.0.0 if we succeed in implementing #36

File: Maybe.Test/MaybeTests.cs (hunk 3 above)
Comment (erickotsuka) [reply Maybe.Test/MaybeTests.cs:359]: I believe implicit cast from type `T?` to type `Maybe<T>`  should not be supported right now given the complications with nullables. Maybe after implementing #36.

Comment (alercunha) [reply Maybe.Test/MaybeTests.cs:359]: I agree with @erickotsuka here. This type of implicit conversion can lead to confusion. We should be explicit in those cases (at least for now).

**Review by d-nery — APPROVED** [review]

**Review by spliner — APPROVED** [review]
